src/main/controllers/ChallengeController.kt

```diff
@@ -12,6 +13,24 @@ import main.services.user_account.ValidateApiKeyService
 import main.services.challenge.*
 import main.helpers.JsonHelper
 
+@Throws(NotFoundException::class)
+fun findChallengeById(challengeId: Int?): Challenge {
```

Comment:
move this to a helper object, can call it ChallengeHelper and just move this function there -- look at the UserAccountHelper for an example.